exe/Main.hs

```diff
@@ -372,10 +381,11 @@ loadSession optTesting ShakeExtras{logger, eventer, restartShakeSession} dir = d
                  let res = (map (renderCradleError ncfp) err, Nothing)
                  modifyVar_ fileToFlags $ \var -> do
                    pure $ Map.insertWith HM.union hieYaml (HM.singleton ncfp (res, dep_info)) var
-                 return res
+                 return (res,[])
 
       -- This caches the mapping from hie.yaml + Mod.hs -> [String]
-      let sessionOpts :: (Maybe FilePath, FilePath) -> IO (IdeResult HscEnvEq)
+      -- Returns the Ghc session and the cradle dependencies
+      let sessionOpts :: (Maybe FilePath, FilePath) -> IO (IdeResult HscEnvEq, [FilePath])
```

Comment:
It looks like there is some duplication here now. We track the modification times outside of Shake via `getDependencyInfo` but we also call `GetModificationTime` on those files. Can we merge the two somehow or remove one of them?

Reply:
That's true, I just didn't have a good enough understanding of the multi cradle code to change it. 

I think the Shake tracking should stay and the cradle code changed to assume that the deps always change (since it wouldn't be called otherwise). But that seems a risky change.

Reply:
I'll give this a quick go and see where it leads, but maybe in a a follow-up PR?

Reply:
Yeah let’s leave it out for now. I’d still like a comment on `sessionVersion`, then this should be good to merge.